### ISSUE TITLE
Modal accessibility - maybe missing aria-modal attribute

### ISSUE THREAD

**Issue by giampierobono** [twbs/bootstrap#22429]
Hello,
  looking into the W3C aria practices repo modal example (hhttps://github.com/w3c/aria-practices/blob/master/examples/dialog-modal/dialog.html) I have noticed that they are using this aria-modal attribute on the element having role=dialog set. 

Reading more about accessible modals here: https://cdn.rawgit.com/w3c/aria-practices/master/aria-practices.html#dialog_modal, it seems that this aria attribute is used when all the elements behind the modal are inert and obscured, instead of setting aria-hidden to all them. 

More details about this attribute here: http://w3c.github.io/aria/aria/aria.html#aria-modal.

Do you think this attribute should be added creating BS modals?

Thanks for you time and your amazing job!

**Comment by patrickhlauke** [comment]
`aria-modal` is new in ARIA 1.1, and to my knowledge not supported yet by any real-world assistive technology. it wouldn't do any harm adding it now, but there would also be no direct benefit just now (though separately i will likely be working on the addition of the `inert` attribute and its polyfill https://github.com/GoogleChrome/inert-polyfill).

**Comment by giampierobono** [comment]
Thanks for your feedback.
I think you are right and currently the aria-modal spec is too draft.
I read somewhere that only a VoiceOver version currently supports it.
